features/0829-VDR-Proxy/README.md

```diff
@@ -0,0 +1,279 @@
+
+#### Server States
+
+The `server` agent goes through the following states:
+
+- request-received
+- completed
+
+The state transition table for the `server` is:
+
+| State / Events   | Receive Request                        | Send Response or Problem Report |
+| ---------------- | -------------------------------------- | ------------------------------- |
+| *Start*          | Transition to <br>**request-received** |                                 |
+| request-received |                                        | Transition to <br>**complete**  |
+| completed        |                                        |                                 |
+
+### Messages
+
+The following are the messages in the VDR Proxy protocol. The `response` message
+handles all completed responses, so the `ack` ([RFC 0015 ACKs]) message is
+**NOT** adopted by this protocol. The [RFC 0035 Report Problem] is adopted by
+this protocol in the event that a `request` is not recognizable as a VDR Proxy
+message and as such, a response message cannot be created. See the
+details below in the [Problem Report Message](#problem-report-message) section.
+
+[RFC 0015 ACKs]: ../features/0015-acks/README.md
+[RFC 0035 Report Problem]: ../features/0035-report-problem/README.md
+
+#### Request Message
+
+The `request` message is sent by the `client` to initiate the protocol. The
+message contains the array of URIs for the `server` to resolve and send the
+resolved objects back to the `client`. It is assumed the `client` knows what
+types of URIs the `server` is prepared to resolve. The `server` returns an error
+status for any URIs it does not know how to resolve. 
+
+The `request` message uses the following format:
+
+```jsonc
+  {
+    "@type": "https://didcomm.org/drpc/1.0/request",
+    "@id": "2a0ec6db-471d-42ed-84ee-f9544db9da4b",
+    "request" : [
+      {
+        "id": "<id>",
+        "uri" : "<uri>"
+      }
+    ]
+  }
+```
+
+The items in the message are as follows:
+
+- `@type` -- required, must be as above
+- `@id` -- required, must be as defined in [RFC 0005 DIDComm]
+- `request` -- **required**, an item containing an array of URIs to be resolved.
+  - `request` **MUST** be an array of requests.
+  - Each request item **MUST** have an `id`, that is unique with the `request` item, typically the index of the array ("1", "2", "3", etc.).
+  - Each request items **MUST** have a URI to be resolved.
+
+On receipt, the server verifies the format of the request, and iterates through the `request` array,
+resolving (or not) each URI and preparing the response message.
+
+#### Response Message
+
+A `response` message is sent by the `server` to following the
+processing of the request to convey the output of the processing to the
+`client`.
+
+If the `request` content is unrecognizable, the `server` SHOULD send a [RFC 0035
+Report Problem] message to the `client`.
+
+```jsonc
+
+  {
+    "@type": "https://didcomm.org/drpc/1.0/response",
```

Comment:
```suggestion
    "@type": "https://didcomm.org/vdrproxy/1.0/response",
```

Reply:
Doh... Thanks.  Can you tell where I started in creating this RFC?  :-)

Reply:
Fixed